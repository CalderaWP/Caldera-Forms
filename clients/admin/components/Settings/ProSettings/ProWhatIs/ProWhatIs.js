import React from 'react';
import propTypes from 'prop-types';
import classNames from 'classnames'

/**
 * Create the ProWhatIs UI
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export const ProWhatIs = (props) => {
	return(
		<div
			className={classNames(props.className,ProWhatIs.classNames.wrapper)}
		>
			<p>Caldera Forms Pro is an app + plugin that makes forms easy.</p>
			<h3>Benefits</h3>
			<ul>
				<li>
					Enhanced Email Delivery
				</li>
				<li>
					Form To PDF
				</li>
				<li>
					Priority Support
				</li>
				<li>
					Add-ons Included in Yearly Plans
				</li>
			</ul>
		</div>
	)
};

/**
 * Prop types for the ProWhatIs component
 * @type {{}}
 */
ProWhatIs.propTypes = {
	classNames: propTypes.string
};


/**
 * Class names used in the ProWhatIs component
 * @type {{wrapper: string}}
 */
ProWhatIs.classNames = {
	wrapper: 'caldera-forms-pro-what-is'
}