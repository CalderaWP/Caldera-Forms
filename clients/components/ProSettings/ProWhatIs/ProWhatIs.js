import React from 'react';
import propTypes from 'prop-types';
import classNames from 'classnames'
import {Twemoji} from 'react-emoji-render';

/**
 * Create the ProWhatIs UI
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export const ProWhatIs = (props) => {
	return(
		<div
			className={classNames(props.className,ProWhatIs.classNames.wrapper)}
		>
			<p className="pro-title">Caldera Forms Pro is an app + plugins that makes forms easy.</p>
			<h3><Twemoji className="emoji" text=":volcano:"/>Benefits</h3>
			<ul>
				<li key="enhanced-mail-delivery">
					<Twemoji className="emoji" text=":email:"/>
					Enhanced Email Delivery
				</li>
				<li key="form-to-pdf">
					<Twemoji className="emoji" text=":eyes:"/>
					Form To PDF
				</li>
				<li key="priority-support">
					<Twemoji className="emoji" text=":volcano:"/>
					Priority Support
				</li>
				<li key="add-ons">
					<Twemoji className="emoji" text=":plug:"/>
					Add-ons Included in Yearly Plans
				</li>
			</ul>
		</div>
	)
};

/**
 * Prop types for the ProWhatIs component
 * @type {{}}
 */
ProWhatIs.propTypes = {
	classNames: propTypes.string
};


/**
 * Class names used in the ProWhatIs component
 * @type {{wrapper: string}}
 */
ProWhatIs.classNames = {
	wrapper: 'caldera-forms-pro-what-is'
}