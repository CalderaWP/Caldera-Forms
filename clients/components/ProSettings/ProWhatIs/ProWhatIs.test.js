import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme/build';
import Enzyme from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import {ProWhatIs} from "./ProWhatIs";

Enzyme.configure({ adapter: new Adapter() });

describe( 'ProWhatIs component', () => {
	it( 'Matches snapshot with minimal props', () => {
		expect(
			renderer.create(
				<ProWhatIs/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'Is wrapped in the right class', () => {
		expect(
			shallow(
				<ProWhatIs/>
			).find( '.' + ProWhatIs.classNames.wrapper )
				.length
		).toEqual(1)
	});
});