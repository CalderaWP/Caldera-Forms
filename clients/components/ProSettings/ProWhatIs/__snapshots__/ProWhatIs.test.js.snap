// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProWhatIs component Matches snapshot with minimal props 1`] = `
<div
  className="caldera-forms-pro-what-is"
>
  <p
    className="pro-title"
  >
    Caldera Forms Pro is an app + plugins that makes forms easy.
  </p>
  <h3>
    <span
      className="emoji undefined"
    >
      <img
        alt="🌋"
        src="http://twemoji.maxcdn.com/2/72x72/1f30b.png"
        style={
          Object {
            "height": "1em",
            "margin": "0 .05em 0 .1em",
            "verticalAlign": "-0.1em",
            "width": "1em",
          }
        }
      />
    </span>
    Benefits
  </h3>
  <ul>
    <li>
      <span
        className="emoji undefined"
      >
        <img
          alt="✉️"
          src="http://twemoji.maxcdn.com/2/72x72/2709.png"
          style={
            Object {
              "height": "1em",
              "margin": "0 .05em 0 .1em",
              "verticalAlign": "-0.1em",
              "width": "1em",
            }
          }
        />
      </span>
      Enhanced Email Delivery
    </li>
    <li>
      <span
        className="emoji undefined"
      >
        <img
          alt="👀"
          src="http://twemoji.maxcdn.com/2/72x72/1f440.png"
          style={
            Object {
              "height": "1em",
              "margin": "0 .05em 0 .1em",
              "verticalAlign": "-0.1em",
              "width": "1em",
            }
          }
        />
      </span>
      Form To PDF
    </li>
    <li>
      <span
        className="emoji undefined"
      >
        <img
          alt="🌋"
          src="http://twemoji.maxcdn.com/2/72x72/1f30b.png"
          style={
            Object {
              "height": "1em",
              "margin": "0 .05em 0 .1em",
              "verticalAlign": "-0.1em",
              "width": "1em",
            }
          }
        />
      </span>
      Priority Support
    </li>
    <li>
      <span
        className="emoji undefined"
      >
        <img
          alt="😛"
          src="http://twemoji.maxcdn.com/2/72x72/1f61b.png"
          style={
            Object {
              "height": "1em",
              "margin": "0 .05em 0 .1em",
              "verticalAlign": "-0.1em",
              "width": "1em",
            }
          }
        />
      </span>
      Add-ons Included in Yearly Plans
    </li>
  </ul>
</div>
`;
